scipy/integrate/_ivp/lsoda.py

```diff
@@ -0,0 +1,195 @@
+        Current status of the solver: 'running', 'finished' or 'failed'.
+    t_crit : float
+        Boundary time.
+    direction : float
+        Integration direction: +1 or -1.
+    t : float
+        Current time.
+    y : ndarray
+        Current state.
+    t_old : float
+        Previous time. None if no steps were made yet.
+    nfev : int
+        Number of the system's rhs evaluations.
+    njev : int
+        Number of the Jacobian evaluations.
+
+    References
+    ----------
+    .. [1] A. C. Hindmarsh, "ODEPACK, A Systematized Collection of ODE
+           Solvers," IMACS Transactions on Scientific Computation, Vol 1.,
+           pp. 55-64, 1983.
+    .. [2] L. Petzold, "Automatic selection of methods for solving stiff and
+           nonstiff systems of ordinary differential equations", SIAM Journal
+           on Scientific and Statistical Computing, Vol. 4, No. 1, pp. 136-148,
+           1983.
+    """
+    def __init__(self, fun, t0, y0, t_crit, first_step=None, min_step=0.0,
+                 max_step=np.inf, rtol=1e-3, atol=1e-6, jac=None, lband=None,
+                 uband=None, vectorized=False, **extraneous):
+        warn_extraneous(extraneous)
+        super(LSODA, self).__init__(fun, t0, y0, t_crit, vectorized)
+
+        if first_step is None:
+            first_step = 0  # LSODA value for automatic selection.
+        elif first_step <= 0:
+            raise ValueError("`first_step` must be positive or None.")
+
+        if max_step == np.inf:
+            max_step = 0  # LSODA value for infinity.
+        elif max_step <= 0:
+            raise ValueError("`max_step` must be positive.")
+
+        if min_step < 0:
+            raise ValueError("`min_step` must be nonnegative.")
+
+        rtol, atol = validate_tol(rtol, atol, self.n)
+
+        self.first_step = first_step
+        self.max_step = max_step
+        self.min_step = min_step
+        self.rtol = rtol
+        self.atol = atol
+        self.jac = jac
+        self.lband = lband
+        self.uband = uband
+
+        solver = ode(self.fun, self.jac)
+        solver.set_integrator('lsoda', rtol=rtol, atol=atol, max_step=max_step,
+                              min_step=min_step, first_step=first_step,
+                              lband=lband, uband=uband)
+        solver.set_initial_value(y0, t0)
+
+        # Inject t_crit into rwork array as needed for itask=5.
+        solver._integrator.rwork[0] = self.t_crit
+        solver._integrator.call_args[4] = solver._integrator.rwork
+
+        self._lsoda_solver = solver
+
+    def _step_impl(self):
+        solver = self._lsoda_solver
+        integrator = solver._integrator
+
+        # From lsoda.step and lsoda.integrate
+        # itask=5 means take a single step and do not go past t_crit.
+        itask = integrator.call_args[2]
+        integrator.call_args[2] = 5
+        solver._y, solver.t = integrator.run(
+            solver.f, solver.jac or (lambda: None), solver._y, solver.t,
```

Comment:
You won't get any argument from me. That section was [copied verbatim from scipy.integrate.ode](https://github.com/scipy/scipy/blob/v0.19.0/scipy/integrate/_ode.py#L406).